packages/server/graphql/intranetSchema/mutations/enableSAMLForDomain.ts

```diff
@@ -19,14 +19,31 @@ const enableSAMLForDomain = {
     const r = await getRethink()
     const normalizedDomain = domain.toLowerCase()
     const normalizedUrl = url.toLowerCase()
-    await r
+
+    const samlDomainExists = (await r
       .table('SAML')
-      .insert({
-        domain: normalizedDomain,
-        url: normalizedUrl,
-        metadata: metadata
-      })
-      .run()
+      .getAll(normalizedDomain, {index: 'domain'})
+      .count()
+      .run()) as number
```

Comment:
it looks like a boolean would be more expressive here, so you could add `.ne(0)`
or if you get the first record, then you can use that id to perform the update